File: Programmers/Python/Level 2/최솟값 만들기/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12941


def solution(A, B):
    answer = 0
    # A의 작은값과 B의 큰값순으로 곱해지도록 정렬
    A.sort()
    B = sorted(B, key=lambda x: -x)
    for a, b in zip(A, B):
        answer += a * b

    return answer

File: Programmers/Javascript/Level 2/최솟값 만들기/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12941?language=javascript#

function solution(A, B) {
  A.sort((a, b) => {
    return a - b;
  });
  B.sort((a, b) => {
    return b - a;
  });

  let answer = 0;
  const length = A.length;

  for (let idx = 0; idx < length; idx++) {
    answer += A[idx] * B[idx];
  }

  return answer;
}
